lib/modules/reducer.js

```diff
@@ -22,7 +22,8 @@ export default (state=DEFAULT, action={}) => {
     }
 
     case actions.SET_PAGE: {
-      const referrerFromHistory = !isEmpty(state.currentPage)
+      const historyLen = state.history.length;
+      const referrerFromHistory = !isEmpty(state.currentPage) && historyLen > 1
```

Comment:
Is it not possible to fix this reducer so that `SET_PAGE` doesn't affect the history on the server side? History is only used by clients so I doubt it would be a huge change...right?